packages/shared/src/session/initializeSession.ts

```diff
@@ -0,0 +1,27 @@
+import { STATUS_RESOLVED } from "../async/createDeferred";
+import { getAuthInfo } from "../authentication/getAuthInfo";
+import { AuthInfo } from "../authentication/types";
+import { deferredAuthInfo, deferredPackageInfo } from "./deferred";
+
+export async function initializeSession({
+  accessToken,
+  packageName,
+  packageVersion,
+}: {
+  accessToken: string | undefined;
+  packageName: string;
+  packageVersion: string;
+}) {
+  if (deferredPackageInfo.status === STATUS_RESOLVED) {
+    return;
+  }
+
+  deferredPackageInfo.resolve({ packageName, packageVersion });
```

Comment:
do we have to make this thing async? This one feels like something that should be available synchronously. But I suppose that:
1. it could complicate the code as we'd have to split `initializeSession` into 2 calls, one that would prefill the package info and the other one that would initialize it with the `accessToken` lazily
2. it might not matter since we need that user agent to make requests... and those usually need to await for the `accessToken` anyway

Reply:
The reason I made this async from an external POV is that I didn't want to assume execution order, particularly for things like our loggers/metrics stuff– which initialize themselves during module parsing (e.g. `new client()`)

This likely happens before the packages call `initializeSession` and so would cause problems.